app/jinja_filters.py

```diff
@@ -279,7 +279,16 @@ class OtherConfig:
     def __init__(self, detail_answer):
         self.id = detail_answer.id
         self.name = detail_answer.name
-        self.value = detail_answer.data or ''
+
+        if detail_answer.data or detail_answer.data == 0:
+            self.value = detail_answer.data
+        elif detail_answer.raw_data and (
+            detail_answer.raw_data[0] or detail_answer.raw_data[0] == 0
+        ):
+            self.value = detail_answer.raw_data[0]
+        else:
+            self.value = ''
+
```

Comment:
This should be consistent with how it is done in the textfield and number templates. All of this can be replaced with `self.value = escape(self._value())`.